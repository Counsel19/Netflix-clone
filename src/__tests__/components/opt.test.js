/* eslint-disable testing-library/no-node-access */
import React from "react";
import { render, screen } from "@testing-library/react";
import { OptForm } from "../../components";

describe("<OptForm>", () => {
  it("renders the <OptForm /> with populated data", () => {
    const { container } = render(
      <OptForm>
        <OptForm.Input placeholder="Email Adress" />
        <OptForm.Button>Try it Now</OptForm.Button>

        <OptForm.Break />

        <OptForm.Text>
          Ready to watch? Enter your email to create or restart your membership
        </OptForm.Text>
      </OptForm>
    );

    expect(screen.getByText('Ready to watch? Enter your email to create or restart your membership')).toBeTruthy();
    expect(screen.getByPlaceholderText('Email Adress')).toBeTruthy();
    expect(screen.getByText('Try it Now')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
