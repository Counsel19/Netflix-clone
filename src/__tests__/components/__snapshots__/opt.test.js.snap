// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptForm> renders the <OptForm /> with populated data 1`] = `
<div
  class="opt-form__Container-sc-q4a5n7-0 itoax"
>
  <input
    class="opt-form__Input-sc-q4a5n7-1 kGRwEn"
    placeholder="Email Adress"
  />
  <button
    class="opt-form__Button-sc-q4a5n7-2 laQfqN"
  >
    Try it Now
    <img
      alt="icon"
      src="./images/icons/chevron-right.png"
    />
  </button>
  <div
    class="opt-form__Break-sc-q4a5n7-4 eCanAT"
  />
  <p
    class="opt-form__Text-sc-q4a5n7-3 eYIOjO"
  >
    Ready to watch? Enter your email to create or restart your membership
  </p>
</div>
`;
